test/NuGet.Core.Tests/NuGet.Build.Tasks.Test/GetReferenceNearestTargetFrameworkTaskTest.cs

```diff
@@ -469,5 +469,161 @@ public void GetReferenceNearestTargetFrameworkTask_WithSingleTfmAndSingleAtfAndN
             testLogger.Errors.Should().Be(1);
             testLogger.DebugMessages.Count.Should().Be(DEBUG_MESSAGE_COUNT_INPUT_OUTPUT);
         }
+
+
+        [Theory]
+        [InlineData(".NETFramework,Version=v4.7.2", "", "net46", ".NETFramework,Version=v4.6", "None", "net46")]
+        [InlineData(".NETCoreApp,Version=v5.0", "Windows,Version=7.0", "net5.0-windows", ".NETCoreApp,Version=v5.0", "Windows,Version=7.0", "net5.0-windows")]
+        [InlineData(".NETCoreApp,Version=v5.0", "Windows,Version=7.0", "net5.0", ".NETCoreApp,Version=v5.0", "None", "net5.0")]
+        [InlineData(".NETCoreApp,Version=v6.0", "Windows,Version=7.0", "net5.0;netcoreapp3.0", ".NETCoreApp,Version=v5.0;.NETCoreApp,Version=v3.0", "None;None", "net5.0")]
+        [InlineData(".NETCoreApp,Version=v6.0", "Android,Version=22.0", "net5.0;net5.0-android", ".NETCoreApp,Version=v5.0;.NETCoreApp,Version=v5.0", "None;android,Version=21.0", "net5.0-android")]
+        [InlineData(".NETCoreApp,Version=v5.0", null, "latestnet;latestnetstandard", ".NETFramework,Version=v4.7.2;.NETStandard,Version=v2.1", "None;None", "latestnetstandard")]
+        public void GetReferenceNearestTargetFrameworkTask_WithTargetFrameworkInformation_ReturnsCompatibleAlias(
```

Comment:
@dsplaisted

Can you please look at this test and confirm this is the contract we agreed on? 

I am gonna more details in the PR body for how it's supposed to work.

Reply:
I think this matches what we discussed.

Reply:
I would probably add a test case for `net5.0` referencing a project multitargeted between `net5.0;net5.0-windows`.

Reply:
Good point, added.